services/core/Darksky/darksky/agent.py

```diff
@@ -0,0 +1,475 @@
+            return "Provides 1 week of daily forecast predictions "\
+                   "by lat/long via RPC (Requires {'lat': <latitude>, 'long': "\
+                   "<longitude>}"
+        else:
+            raise RuntimeError(
+                "Service {} is not implemented by Darksky.".format(
+                    service_name))
+
+    def get_point_name_defs_file(self):
+        """
+        Constructs the point name mapping dict from the
+        mapping csv.
+        :return: dictionary containing a mapping of service point
+        names to standard point names with optional
+        """
+        return pkg_resources.resource_stream(__name__, "data/name_mapping.csv")
+
+    def get_darksky_forecast(self, service, location):
+        """
+        Generic method called by the current and forecast service endpoint
+        methods to fetch a forecast request from the Darksky API. If
+        performance mode is set to True, the url adds exclusions for the
+        services provided by the API that were not requested.
+        :param service: requested service endpoint
+        :param location: location dictionary for building url
+        :return: Darksky forecast request response
+        """
+        service_json_name = ''
+        for service_code in SERVICES_MAPPING:
+            if SERVICES_MAPPING[service_code]['service'] is service:
+                service_json_name = SERVICES_MAPPING[service_code]['json_name']
+        if "lat" in location and 'long' in location:
+            url = "https://api.darksky.net/forecast/{key}/{lat}," \
+               "{long}?units=si".format(key=self._api_key, lat=location['lat'],
+                                        long=location['long'])
+            if self.performance_mode:
+                services = ["currently", "hourly", "minutely", "daily"]
+                if service_json_name and service_json_name in services:
+                    services.remove(service_json_name)
+                    url += "&exclude=" + ",".join(services)
+                else:
+                    raise RuntimeError("Requested service {} is not provided"
+                                       " by the Darksky API".format(service))
+        else:
+            raise ValueError('Invalid location. Expected format is: '
+                             '"{"lat": "xxx.xxxx", "long": "xxx.xxxx"}"')
+        grequest = [grequests.get(url, verify=requests.certs.where(),
+                                  headers=self.headers, timeout=3)]
+        gresponse = grequests.map(grequest)[0]
+        if gresponse is None:
+            raise RuntimeError("get request did not return any "
+                               "response")
+        try:
+            response = json.loads(gresponse.content)
+            return response
+        except ValueError:
+            self.generate_response_error(url, gresponse.status_code)
+
+    def format_multientry_response(self, location, response, service):
```

Comment:
Why do we manipulate generation_time for forecast instead of returning just current time ? In base weather both get_remote_forecast and get_remote_current - we simply convert current time into UTC and save it to db in the right format. Why is it different for forecast of darksky ?

Reply:
Generation time is the time the predicted values were generated by the api -  this is api dependent, so we should use their scheme.